{
  checkState(!shutdown,"shutdown!");
  checkArgument(tasks.size() > 0,"no tasks");
  Exception exception=null;
  for (  Callable<T> task : tasks) {
    checkNotNull(task,"task");
    try {
      return task.call();
    }
 catch (    InterruptedException e) {
      throw e;
    }
catch (    Exception e) {
      exception=e;
    }
  }
  throw new ExecutionException("no tasks completed successfully",exception);
}
